### PR TITLE
Add ValenciaDivClean Outflow BC

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Adds the outflow `BoundaryCondition` to the `ValenciaDivClean` system -- this boundary condition just checks that the characteristic speeds are directed out of the domain, without altering any of the evolved variables or their time derivatives.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      const tnsr::I<DataVector, 3, Frame::Inertial>& spatial_velocity,
+      const Scalar<DataVector>& lorentz_factor,
+      const tnsr::I<DataVector, 3, Frame::Inertial>& magnetic_field,
+
+      const tnsr::I<DataVector, 3, Frame::Inertial>& shift,
+      const Scalar<DataVector>& lapse,
+      const tnsr::II<DataVector, 3, Frame::Inertial>& inverse_spatial_metric,
+
+      const EquationOfState& equation_of_state) noexcept {
+    double min_speed = std::numeric_limits<double>::signaling_NaN();
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp]: aren't the char speeds bounded by `\pm \alpha -\beta^i n_i` and so you only need to check those?

Comment (moxcodes) [reply src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp]: Ah, yep, seems correct... That'll simplify matters considerably.

**Comment by moxcodes** [comment]
The change shrunk the size of the PR by a factor of 2, so I've squashed directly for readability.

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+      const tnsr::I<DataVector, 3, Frame::Inertial>& shift,
+      const Scalar<DataVector>& lapse) noexcept {
+    double min_speed = std::numeric_limits<double>::signaling_NaN();
+    Variables<tmpl::list<::Tags::TempScalar<0>, ::Tags::TempScalar<1>>> buffer{
+        get(lapse).size()};
+    auto& normal_dot_shift = get<::Tags::TempScalar<0>>(buffer);
+    dot_product(make_not_null(&normal_dot_shift),
+                outward_directed_normal_covector, shift);
+    auto& normal_dot_mesh_velocity = get<::Tags::TempScalar<1>>(buffer);
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp]: I think this is only used inside the `if`, right? If so, can it be scoped into there?

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  void pup(PUP::er& p) override;
+
+  using dg_interior_evolved_variables_tags = tmpl::list<>;
+  using dg_interior_temporary_tags =
+      tmpl::list<gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+                 gr::Tags::Lapse<DataVector>>;
+  using dg_gridless_tags = tmpl::list<>;
+  using dg_interior_primitive_variables_tags = tmpl::list<>;
+
+  static std::optional<std::string> dg_outflow(
```
Comment (nilsdeppe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Outflow.hpp:61]: what do you think of moving this function definition into the cpp file? I don't think there's anything preventing that

**Review by nilsdeppe — DISMISSED** [review]
LGTM, two small suggestions you can squash immediately if you decide to implement them :)

**Comment by moxcodes** [comment]
fixes squashed in.
Cheers!

**Review by nilsdeppe — DISMISSED** [review]

**Comment by moxcodes** [comment]
updated includes.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**